### DISCUSSION TITLE
ambiguous documentation prisma-client-reference

### DISCUSSION THREAD

**Discussion by ranran2121** [prisma/prisma#17046] [Q&A]
Hi, in the prisma-client-reference documentation ([https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#disconnect](url), I read the following regarding the upsert
```
Update an existing User record by updating two Post record it's connected to or creating new ones (upsert):

const user = await prisma.user.update({
  where: { email: 'alice@prisma.io' },
  data: {
    posts: {
      upsert: [
        {
          create: { title: 'This is my first post' },
          update: { title: 'This is my first post' },
          where: { id: 32 },
        },
        {
          create: { title: 'This is my second post' },
          update: { title: 'This is my second post' },
          where: { id: 23 },
        },
      ],
    },
  },
})
```
however, If I am creating a post, I do not have the id to write in the ```where``` key. 
How do you manage that?

**Comment by nurul3101** [comment]
Hello @ranran2121 👋 

The main idea behind upsert is that based on the where clause PrismaClient identifies if a record needs to be inserted or updated. If a matching record based on the where clause is identified, then the `update` clause will be executed and if no records are found, then the `create` clause will be executed: [Reference](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#upsert).

Does that make sense, or I am misunderstanding something?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
